dotCMS/src/test/java/com/dotcms/filters/NormalizationFilterTest.java

```diff
@@ -0,0 +1,122 @@
+        doNothing().when(chain).doFilter(capturedRequest.capture(), any(HttpServletResponse.class));
+    }
+
+    /**
+     * Test to verify the {@link NormalizationFilter} is applying properly the normalization on
+     * invalids URIs
+     */
+    @Test
+    public void test_uri_normalization_invalid_URI() throws IOException, ServletException {
+
+        // A ".." segment is removed only if it is preceded by a non-".." segment
+        String originalURI = "/test/../folder/important/secret_file.dat";
+        String expectedNormalizedURI = "/folder/important/secret_file.dat";
+        compare(originalURI, expectedNormalizedURI, Boolean.FALSE);
+
+        // A ".." segment is removed only if it is preceded by a non-".." segment
+        originalURI = "/test/../folder/folder1/forward_jsp.jsp?FORWARD_URL=http://google.com";
+        expectedNormalizedURI = "/folder/folder1/forward_jsp.jsp?FORWARD_URL=http://google.com";
+        compare(originalURI, expectedNormalizedURI, Boolean.FALSE);
+
+        // A ".." segment is removed only if it is preceded by a non-".." segment
+        originalURI = "/test/../folder/important/../secret_file.dat";
+        expectedNormalizedURI = "/folder/secret_file.dat";
+        compare(originalURI, expectedNormalizedURI, Boolean.FALSE);
+
+        // Each "." segment is simply removed
+        originalURI = "./folder/folder1/file.dat";
+        expectedNormalizedURI = "folder/folder1/file.dat";
+        compare(originalURI, expectedNormalizedURI, Boolean.FALSE);
+
+        // Each "." segment is simply removed
+        originalURI = "./folder/./folder1/file.dat";
+        expectedNormalizedURI = "folder/folder1/file.dat";
+        compare(originalURI, expectedNormalizedURI, Boolean.FALSE);
+
+        // Each "." segment is simply removed
+        originalURI = "/folder/./folder1/file.dat";
+        expectedNormalizedURI = "/folder/folder1/file.dat";
+        compare(originalURI, expectedNormalizedURI, Boolean.FALSE);
+    }
+
+    /**
+     * Test to verify the {@link NormalizationFilter} is applying properly the normalization on
+     * valid URIs
+     */
+    @Test
+    public void test_uri_normalization_valid_URI() throws IOException, ServletException {
+
+        String originalURI = "/folder/important/secret_file.dat";
+        compare(originalURI, originalURI, Boolean.TRUE);
+
+        originalURI = "/folder/folder1/forward_jsp.jsp?FORWARD_URL=http://google.com";
+        compare(originalURI, originalURI, Boolean.TRUE);
+
+        originalURI = "folder/folder1/file.dat";
+        compare(originalURI, originalURI, Boolean.TRUE);
+
+        // A ".." segment is removed only if it is preceded by a non-".." segment
+        originalURI = "../folder/folder1/file.dat";
+        compare(originalURI, originalURI, Boolean.TRUE);
+    }
+
+    private void compare(final String originalURI, final String expectedNormalizedURI,
+            boolean equals)
+            throws IOException, ServletException {
```

Comment:
Is the third param really needed? Seems like should be enough with comparing equality between the first two  (originalURI,  expectedNormalizedURI)